test/success/connection.prl

```diff
@@ -7,13 +7,20 @@ Thm Connection(#A) : [
 ] by [
   {
   lam a. lam b. lam p.
-    <i> <j> 'hcom{i=0,i=1,j=0,j=1;0~>1}(#A; ,a;
-     {k}.hcom{k=0,k=1;1~>j}(#A; hcom{k=0,k=1;1~>0}(#A; ,p @ k; {_}.,a; {j}.,p @ j); {_}.,a; {_}.,a);
-     {k}.hcom{k=0,k=1;1~>j}(#A; ,p @ k; {_}.,a; {j}.,p @ j);
-     {k}.hcom{k=0,k=1;1~>i}(#A; hcom{k=0,k=1;1~>0}(#A; ,p @ k; {_}.,a; {i}.,p @ i); {_}.,a; {_}.,a);
-     {k}.hcom{k=0,k=1;1~>i}(#A; ,p @ k; {_}.,a; {j}.,p @ j))
+    <i> <j> 
+      'hcom{0~>1}(#A; ,a)
+        [i=0 -> {k}.
+          hcom{1~>j}(#A; hcom{1~>0}(#A; ,p @ k) [k=0 -> {_}.,a; k=1 -> {l}.,p @ l])
+            [k=0 -> {_}.,a;
+             k=1 -> {_}.,a];
+         i=1 -> {k}. hcom{1~>j}(#A; ,p @ k) [k=0 -> {_}.,a; k=1 -> {l}.,p @ l];
+         j=0 -> {k}.
+          hcom{1~>i}(#A; hcom{1~>0}(#A; ,p @ k) [k=0 -> {_}.,a; k=1 -> {l}.,p @ l])
+            [k=0 -> {_}.,a;
+             k=1 -> {_}.,a];
+         j=1 -> {k}. hcom{1~>i}(#A; ,p @ k) [k=0 -> {_}.,a; k=1 -> {l}.,p @ l]]
   };
-  auto;
+  auto
```

Comment:
I did some study but decided not to spend too much time now. You might be interested in this:
```
0.31 seconds of CPU time (0.13 seconds GC)
            function              cur 
-------------------------------- -----
<gc>                             29.5%
Telescope.Internal.ConsView.out  22.7%
Susp.delay                        6.8%
Abt.check                         4.5%
Abt.inferb                        4.5%
Abt.infer                         2.3%
Abt.symctx                        2.3%
Abt.liberateSymbolAnn             2.3%
Abt.liberateVariable.fn           2.3%
Abt.foldStar.fn.fn                2.3%
Abt.ShiftFunCat.cmp               2.3%
Abt.assertSortEq                  2.3%
Susp.force                        2.3%
DictFun.union                     2.3%
SplayRDict.foldl                  2.3%
AbtParameterTerm.Monad.bind.fn    2.3%
SplayTree.findAndSplay            2.3%
RedPrlCategoricalJudgment.map.fn  2.3%
Sequence.append                   2.3%
```